pkgs/top-level/all-packages.nix

```diff
@@ -37516,11 +37516,6 @@ with pkgs;
 
   tcl2048 = callPackage ../games/tcl2048 { };
 
-  the-powder-toy = callPackage ../games/the-powder-toy {
-    lua = lua5_1;
-    inherit (darwin.apple_sdk.frameworks) Cocoa;
-  };
-
```

Comment:
This should not be removed, rather the path should be changed to `../by-name/th/the-powder-toy/package.nix`

(Though it now uses lua 5.2, so should likely be pinned to that.)

Reply:
Why is this? The line at https://github.com/NixOS/nixpkgs/blob/9863d8c74a959d73df0c495a55097f804f7b379d/pkgs/by-name/th/the-powder-toy/package.nix#L2 means the package behaves identical after the update as it did before (in regards to .override which as far as i understood is the motivation) even without listing in all-packages.nix, so why should it still be listed in all-packages.nix?

Reply:
lua versions aren't backwards-compatible
and in that case what's the point of having Cocoa as a function arg?

Reply:
I have no idea why Cocoa is a function arg, it was there before i touched the module and figured getting rid of it would break the existing override behaviour. I'd be more than happy getting rid of it.